File: ios/Editor/DesignTokens.swift
import UIKit

/// Default values for the "Studio Violet" design system used by the native editor
/// shell, matching the Stitch-exported mockups under
/// stitch_react_native_photo_editor_sdk/studio_violet_creative_sdk/DESIGN.md.
/// Mirrors `DesignTokens.kt`. These are FALLBACKS only — every color a consumer
/// can already override via `EditorOptions.theme` still wins; this object just
/// changes what the editor looks like when no override is given.
enum DesignTokens {
  // Surface hierarchy
  static let surfaceDim = UIColor(red: 0x13 / 255, green: 0x13 / 255, blue: 0x16 / 255, alpha: 1)
  static let surface = surfaceDim
  static let surfaceContainerLowest = UIColor(red: 0x0E / 255, green: 0x0E / 255, blue: 0x11 / 255, alpha: 1)
  static let surfaceContainerLow = UIColor(red: 0x1B / 255, green: 0x1B / 255, blue: 0x1E / 255, alpha: 1)
  static let surfaceContainer = UIColor(red: 0x1F / 255, green: 0x1F / 255, blue: 0x22 / 255, alpha: 1)
  static let surfaceContainerHigh = UIColor(red: 0x2A / 255, green: 0x2A / 255, blue: 0x2D / 255, alpha: 1)
  static let surfaceContainerHighest = UIColor(red: 0x35 / 255, green: 0x34 / 255, blue: 0x38 / 255, alpha: 1)
  static let surfaceVariant = surfaceContainerHighest
  static let surfaceBright = UIColor(red: 0x39 / 255, green: 0x39 / 255, blue: 0x3C / 255, alpha: 1)

  // Content
  static let onSurface = UIColor(red: 0xE4 / 255, green: 0xE1 / 255, blue: 0xE6 / 255, alpha: 1)
  static let onSurfaceVariant = UIColor(red: 0xCC / 255, green: 0xC3 / 255, blue: 0xD8 / 255, alpha: 1)
  static let outline = UIColor(red: 0x95 / 255, green: 0x8D / 255, blue: 0xA1 / 255, alpha: 1)
  static let outlineVariant = UIColor(red: 0x4A / 255, green: 0x44 / 255, blue: 0x55 / 255, alpha: 1)

  // Accents
  static let primary = UIColor(red: 0xD2 / 255, green: 0xBB / 255, blue: 0xFF / 255, alpha: 1)
  static let onPrimary = UIColor(red: 0x3F / 255, green: 0x00 / 255, blue: 0x8E / 255, alpha: 1)
  static let primaryContainer = UIColor(red: 0x7C / 255, green: 0x3A / 255, blue: 0xED / 255, alpha: 1)
  static let onPrimaryContainer = UIColor(red: 0xED / 255, green: 0xE0 / 255, blue: 0xFF / 255, alpha: 1)
  static let secondary = UIColor(red: 0xC0 / 255, green: 0xC1 / 255, blue: 0xFF / 255, alpha: 1)
  static let tertiary = UIColor(red: 0xD0 / 255, green: 0xBC / 255, blue: 0xFF / 255, alpha: 1)

  // Literal Stitch accent hexes (DESIGN.md "Primary Accents & Active States"), used by the
  // new EditorUI components where the exact mockup hue matters more than the M3 mapping.
  static let accentViolet = UIColor(red: 0x7C / 255, green: 0x3A / 255, blue: 0xED / 255, alpha: 1)
  static let accentIndigo = UIColor(red: 0x63 / 255, green: 0x66 / 255, blue: 0xF1 / 255, alpha: 1)
  static let accentHighlightViolet = UIColor(red: 0x8B / 255, green: 0x5C / 255, blue: 0xF6 / 255, alpha: 1)

  // Literal Stitch surface hexes (DESIGN.md "Canvas & Surface Hierarchy").
  static let canvasViewport = UIColor(red: 0x09 / 255, green: 0x09 / 255, blue: 0x0B / 255, alpha: 1)
  static let editorSurface = UIColor(red: 0x12 / 255, green: 0x12 / 255, blue: 0x14 / 255, alpha: 1)
  static let elevatedPanel = UIColor(red: 0x18 / 255, green: 0x18 / 255, blue: 0x1B / 255, alpha: 1)
  static let interactiveNeutral = UIColor(red: 0x27 / 255, green: 0x27 / 255, blue: 0x2A / 255, alpha: 1)
  static let mutedBorderColor = UIColor(red: 0x3F / 255, green: 0x3F / 255, blue: 0x46 / 255, alpha: 1)

  // Literal Stitch text hexes (DESIGN.md "Typography & Content Grayscale").
  static let textPrimary = UIColor.white
  static let textSecondary = UIColor(red: 0xA1 / 255, green: 0xA1 / 255, blue: 0xAA / 255, alpha: 1)
  static let textSubtle = UIColor(red: 0x71 / 255, green: 0x71 / 255, blue: 0x7A / 255, alpha: 1)

  // Spacing (points), matching DESIGN.md's `space-*` scale.
  static let spaceXs: CGFloat = 4
  static let spaceSm: CGFloat = 8
  static let spaceMd: CGFloat = 12
  static let spaceLg: CGFloat = 16
  static let spaceXl: CGFloat = 20
  static let space2xl: CGFloat = 24
  static let space3xl: CGFloat = 32
  static let touchTargetMin: CGFloat = 44

  // Corner radii (points)
  static let radiusSm: CGFloat = 4
  static let radiusMd: CGFloat = 8
  static let radiusLg: CGFloat = 12
  static let radiusXl: CGFloat = 16
  static let radiusSheet: CGFloat = 24

  // Elevation tiers (DESIGN.md "Elevation & Depth"): background color + opacity to composite
  // persistent chrome / floating trays / popovers over the canvas.
  enum ElevationTier1 { static let color = editorSurface; static let opacity: CGFloat = 0.92 }
  enum ElevationTier2 { static let color = elevatedPanel; static let opacity: CGFloat = 0.88 }
  enum ElevationTier3 { static let color = interactiveNeutral; static let opacity: CGFloat = 1.0 }
}

File: android/src/main/java/com/photovideoeditor/ui/DesignTokens.kt
package com.photovideoeditor.ui

import android.graphics.Color

/**
 * Default values for the "Studio Violet" design system used by the native editor
 * shell, matching the Stitch-exported mockups under
 * stitch_react_native_photo_editor_sdk/studio_violet_creative_sdk/DESIGN.md.
 * These are FALLBACKS only — every color a consumer can already override via
 * `EditorOptions.theme` (see `PhotoVideoEditorActivity.themeColor`) still wins;
 * this object just changes what the editor looks like when no override is given.
 */
object DesignTokens {
  // Surface hierarchy
  const val surfaceDim = 0xFF131316.toInt()
  const val surface = 0xFF131316.toInt()
  const val surfaceContainerLowest = 0xFF0E0E11.toInt()
  const val surfaceContainerLow = 0xFF1B1B1E.toInt()
  const val surfaceContainer = 0xFF1F1F22.toInt()
  const val surfaceContainerHigh = 0xFF2A2A2D.toInt()
  const val surfaceContainerHighest = 0xFF353438.toInt()
  const val surfaceVariant = 0xFF353438.toInt()
  const val surfaceBright = 0xFF39393C.toInt()

  // Content
  const val onSurface = 0xFFE4E1E6.toInt()
  const val onSurfaceVariant = 0xFFCCC3D8.toInt()
  const val outline = 0xFF958DA1.toInt()
  const val outlineVariant = 0xFF4A4455.toInt()

  // Accents
  const val primary = 0xFFD2BBFF.toInt()
  const val onPrimary = 0xFF3F008E.toInt()
  const val primaryContainer = 0xFF7C3AED.toInt()
  const val onPrimaryContainer = 0xFFEDE0FF.toInt()
  const val secondary = 0xFFC0C1FF.toInt()
  const val tertiary = 0xFFD0BCFF.toInt()
  const val error = 0xFFFFB4AB.toInt()
  const val errorContainer = 0xFF93000A.toInt()

  // Literal Stitch accent hexes (DESIGN.md "Primary Accents & Active States"), used by the
  // new EditorUI components below where the exact mockup hue matters more than the M3 mapping.
  const val accentViolet = 0xFF7C3AED.toInt()
  const val accentIndigo = 0xFF6366F1.toInt()
  const val accentHighlightViolet = 0xFF8B5CF6.toInt()

  // Literal Stitch surface hexes (DESIGN.md "Canvas & Surface Hierarchy").
  const val canvasViewport = 0xFF09090B.toInt()
  const val editorSurface = 0xFF121214.toInt()
  const val elevatedPanel = 0xFF18181B.toInt()
  const val interactiveNeutral = 0xFF27272A.toInt()
  const val mutedBorderColor = 0xFF3F3F46.toInt()

  // Literal Stitch text hexes (DESIGN.md "Typography & Content Grayscale").
  const val textPrimary = 0xFFFFFFFF.toInt()
  const val textSecondary = 0xFFA1A1AA.toInt()
  const val textSubtle = 0xFF71717A.toInt()

  // Backwards-compatible aliases matching the flatter names `themeColor()` fallbacks used before this pass.
  const val toolbarColor = surfaceContainerLow
  const val backgroundColor = surfaceContainerLowest
  const val textColor = onSurface
  const val primaryColor = primaryContainer

  /** [color] at [alphaPercent]% opacity (0..100), matching the mockups' translucent surface tiers. */
  fun withAlphaPercent(color: Int, alphaPercent: Int): Int {
    val alpha = ((alphaPercent.coerceIn(0, 100)) * 255 / 100)
    return Color.argb(alpha, Color.red(color), Color.green(color), Color.blue(color))
  }

  // Spacing (dp), matching DESIGN.md's `space-*` scale.
  const val spaceXs = 4
  const val spaceSm = 8
  const val spaceMd = 12
  const val spaceLg = 16
  const val spaceXl = 20
  const val space2xl = 24
  const val space3xl = 32
  const val touchTargetMin = 44

  // Corner radii (dp)
  const val radiusSm = 4
  const val radiusMd = 8
  const val radiusLg = 12
  const val radiusXl = 16
  const val radiusSheet = 24

  // Elevation tiers (DESIGN.md "Elevation & Depth"): background color + opacity percent
  // to composite persistent chrome / floating trays / popovers over the canvas.
  object ElevationTier1 { const val color = editorSurface; const val opacityPercent = 92 }
  object ElevationTier2 { const val color = elevatedPanel; const val opacityPercent = 88 }
  object ElevationTier3 { const val color = interactiveNeutral; const val opacityPercent = 100 }
}
